src/BenchmarkDotNet/Engines/EnginePilotStage.cs

```diff
@@ -67,7 +68,10 @@ private long RunAuto()
                 if (invokeCount >= MaxInvokeCount)
                     break;
 
-                invokeCount *= 2;
+                if (unrollFactor == 1 && invokeCount < EnvResolver.DefaultUnrollFactorForThroughput)
```

Comment:
@AndreyAkinshin the `invokeCount < EnvResolver.DefaultUnrollFactorForThroughput` check is for the benchmarks, where the first call is super expensive and every next can be very fast (like initiating the cache and later just reading from it)

So we are going to increment it until we reach `EnvResolver.DefaultUnrollFactorForThroughput` and after it just double it.